plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeComputedStatistics.java

```diff
@@ -0,0 +1,102 @@
+import static io.trino.spi.type.VarbinaryType.VARBINARY;
+
+public final class DeltaLakeComputedStatistics
+{
+    private DeltaLakeComputedStatistics() {}
+
+    public static DeltaLakeJsonFileStatistics toDeltaLakeJsonFileStatistics(ComputedStatistics stats, Map<String, DeltaLakeColumnHandle> columnHandles)
+    {
+        Optional<Long> rowCount = getLongValue(stats.getTableStatistics().get(ROW_COUNT)).stream().boxed().findFirst();
+
+        Optional<Map<String, Object>> maxValues = Optional.of(getColumnStatistics(stats, MAX_VALUE, columnHandles));
+        Optional<Map<String, Object>> minValues = Optional.of(getColumnStatistics(stats, MIN_VALUE, columnHandles));
+
+        Optional<Map<String, Object>> nullCount = Optional.empty();
+        if (rowCount.isPresent()) {
+            nullCount = Optional.of(getNullCount(stats, rowCount.get(), columnHandles));
+        }
+
+        return new DeltaLakeJsonFileStatistics(rowCount, minValues, maxValues, nullCount);
+    }
+
+    private static Map<String, Object> getNullCount(ComputedStatistics statistics, long rowCount, Map<String, DeltaLakeColumnHandle> columnHandles)
+    {
+        return statistics.getColumnStatistics().entrySet().stream()
+                .filter(stats -> stats.getKey().getStatisticType().equals(NUMBER_OF_NON_NULL_VALUES)
+                        && columnHandles.containsKey(stats.getKey().getColumnName()))
+                .map(stats -> Map.entry(columnHandles.get(stats.getKey().getColumnName()).getBasePhysicalColumnName(), getLongValue(stats.getValue())))
+                .filter(stats -> stats.getValue().isPresent())
+                .map(nonNullCount -> Map.entry(nonNullCount.getKey(), rowCount - nonNullCount.getValue().getAsLong()))
+                .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private static Map<String, Object> getColumnStatistics(ComputedStatistics statistics, ColumnStatisticType statisticType, Map<String, DeltaLakeColumnHandle> columnHandles)
+    {
+        return statistics.getColumnStatistics().entrySet().stream()
+                .filter(stats -> stats.getKey().getStatisticType().equals(statisticType)
+                        && columnHandles.containsKey(stats.getKey().getColumnName()))
+                .map(stats -> mapSingleStatisticsValueToJsonRepresentation(stats, columnHandles))
+                .filter(Optional::isPresent)
+                .flatMap(Optional::stream)
+                .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private static Optional<Map.Entry<String, Object>> mapSingleStatisticsValueToJsonRepresentation(Map.Entry<ColumnStatisticMetadata, Block> statistics, Map<String, DeltaLakeColumnHandle> columnHandles)
+    {
+        Type columnType = columnHandles.get(statistics.getKey().getColumnName()).getBasePhysicalType();
+        String physicalName = columnHandles.get(statistics.getKey().getColumnName()).getBasePhysicalColumnName();
```

Comment:
This is actually a canonical physical name. When a column mapping mode is none and the column contains uppercase characters (this will happen if the table is created by Spark), this logic results in lowercase names.

Reply:
After rebase (including https://github.com/trinodb/trino/pull/18123) `physicalName` can have mixed case if created as such.